flash/vision/classification/data.py

```diff
@@ -176,204 +63,316 @@ def _find_classes(self, dir):
         class_to_idx = {cls_name: i for i, cls_name in enumerate(classes)}
         return classes, class_to_idx
 
-    def __getitem__(self, index):
-        """
-        Args:
-            index (int): Index
+    @staticmethod
+    def _get_predicting_files(samples):
+        files = []
+        if isinstance(samples, str):
+            samples = [samples]
 
-        Returns:
-            tuple: (sample, target) where target is class_index of the target class.
-        """
-        if self.with_targets:
-            path, target = self.samples[index]
-            if self.target_transform is not None:
-                target = self.target_transform(target)
+        if isinstance(samples, list) and all(os.path.isdir(s) for s in samples):
+            for s in samples:
+                for f in os.listdir(s):
+                    files += [os.path.join(s, f)]
+
+        elif isinstance(samples, list) and all(os.path.isfile(s) for s in samples):
+            files = samples
+
+        files = list(filter(lambda p: has_file_allowed_extension(p, IMG_EXTENSIONS), files))
+
+        return files
+
+    @classmethod
+    def load_data(cls, samples: Any, dataset: Optional[AutoDataset] = None) -> Any:
+        classes, class_to_idx = cls._find_classes(samples)
+        dataset.num_classes = len(classes)
+        return make_dataset(samples, class_to_idx, IMG_EXTENSIONS, None)
```

Comment:
@tchaton @kaushikb11 @justusschock  I found here somehow an inconsistency when using the API `from_folders` where the train folder is `root/train` (as I understand the API should be used) but `make_dataset` assumes just the root path. Then this returns an empty list and make the rest of the pipeline of loading data to crash.

Reply:
I won't say it is an inconsistency, but more a missing features. We were just following TorchVision API.